runtime/bcutil/defineclass.c

```diff
@@ -871,6 +871,20 @@ createROMClassFromClassFile(J9VMThread *currentThread, J9LoadROMClassData *loadD
 			exceptionNumber = J9VMCONSTANTPOOL_JAVALANGOUTOFMEMORYERROR;
 			break;
 
+		case BCT_ERR_GENERIC_ERROR_CUSTOM: {
+			/* default value for exceptionNumber (J9VMCONSTANTPOOL_JAVALANGCLASSFORMATERROR) assigned before switch */
+			/* custom message that should be combined with the faulty class name */
+			UDATA messageSize = strlen((char*)vm->dynamicLoadBuffers->classFileError);
+			UDATA memorySize = messageSize + loadData->classNameLength + 1;
+
+			PORT_ACCESS_FROM_JAVAVM(vm);
+			errorUTF = j9mem_allocate_memory(memorySize, J9MEM_CATEGORY_CLASSES);
+			memcpy(errorUTF, vm->dynamicLoadBuffers->classFileError, messageSize);
+			memcpy(errorUTF + messageSize, loadData->className, loadData->classNameLength);
+			errorUTF[memorySize - 1] = (U_8) '\0';
```

Comment:
is that possible to call `buildVerifyErrorString` to customize the detailed error message like the case at BCT_ERR_CLASS_READ
```
	switch (result) {
		case BCT_ERR_CLASS_READ: {
			J9CfrError * cfrError = (J9CfrError *) vm->dynamicLoadBuffers->classFileError;
			errorUTF = (U_8 *) buildVerifyErrorString(vm, cfrError, className, classNameLength);
			exceptionNumber = cfrError->errorAction;
			break;
		}
```

Reply:
i'll try that